components/code_preview.jsx

```diff
@@ -24,28 +24,46 @@ export default class CodePreview extends React.PureComponent {
     }
 
     componentDidMount() {
-        this.updateStateFromProps(this.props);
+        this.getCode();
     }
 
-    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
-        if (this.props.fileUrl !== nextProps.fileUrl) {
-            this.updateStateFromProps(nextProps);
+    static getDerivedStateFromProps(props, state) {
```

Comment:
Since we are removing old lifecycle method do you think this is a good candidate for using React Hooks?

Reply:
I don't think we need to do that now. We can create a followup ticket to do that. For now, I just want to get rid of the deprecated lifecycle methods in favour of the new ones.